libsolutil/FixedHash.h

```diff
@@ -64,10 +64,10 @@ class FixedHash
 	FixedHash(Arith const& _arith) { toBigEndian(_arith, m_data); }
 
 	/// Explicitly construct, copying from a byte array.
-	explicit FixedHash(bytes const& _b, ConstructFromHashType _t = FailIfDifferent) { if (_b.size() == N) memcpy(m_data.data(), _b.data(), std::min<unsigned>(_b.size(), N)); else { m_data.fill(0); if (_t != FailIfDifferent) { auto c = std::min<unsigned>(_b.size(), N); for (unsigned i = 0; i < c; ++i) m_data[_t == AlignRight ? N - 1 - i : i] = _b[_t == AlignRight ? _b.size() - 1 - i : i]; } } }
+	explicit FixedHash(bytes const& _b, ConstructFromHashType _t = FailIfDifferent) { if (_b.size() == N) memcpy(m_data.data(), _b.data(), std::min<size_t>(_b.size(), N)); else { m_data.fill(0); if (_t != FailIfDifferent) { auto c = std::min<size_t>(_b.size(), N); for (size_t i = 0; i < c; ++i) m_data[_t == AlignRight ? N - 1 - i : i] = _b[_t == AlignRight ? _b.size() - 1 - i : i]; } } }
```

Comment:
Do you know if size of `_b` will always fit inside `unsigned`?

Reply:
I didn't change any logic except using proper min functions.
Where exactly you think there is a problem

Reply:
Can't this line:
```cpp
if (_b.size() == N) memcpy(m_data.data(), _b.data(), std::min<unsigned>(_b.size(), N));
```
be rewritten to

```cpp
if (_b.size() == N) memcpy(m_data.data(), _b.data(), size_t(N));
```

Why do we do a min after checking for equality? Or I don't understand this code.


------
My previous question doesn't make sense anymore.

Reply:
completely agree :) When it was one line it was really hard to see the context
Fixing like you suggested

Reply:
@mijovic Do you know if we have tests for these hashing?

Reply:
I really don’t know for tests
We should check and open issues if there are no tests